tests/test_ssm/test_ssm_boto3.py

```diff
@@ -394,19 +407,42 @@ def test_put_parameter_china():
     response["Version"].should.equal(1)
 
 
+@mock_ssm
+@pytest.mark.parametrize("bad_data_type", ["not_text", "not_ec2", "something weird"])
+def test_put_parameter_invalid_data_type(bad_data_type):
+    client = boto3.client("ssm", region_name="us-east-1")
+    with pytest.raises(ClientError) as e:
+        client.put_parameter(
+            Name="test_name", Value="some_value", Type="String", DataType=bad_data_type
+        )
+    ex = e.value
+    ex.operation_name.should.equal("PutParameter")
+    ex.response["ResponseMetadata"]["HTTPStatusCode"].should.equal(400)
+    ex.response["Error"]["Code"].should.contain("ValidationException")
+    ex.response["Error"]["Message"].should.equal(
+        f"The following data type is not supported: {bad_data_type}"
+        " (Data type names are all lowercase.)"
+    )
+
+
 @mock_ssm
 def test_get_parameter():
     client = boto3.client("ssm", region_name="us-east-1")
 
     client.put_parameter(
-        Name="test", Description="A test parameter", Value="value", Type="String"
+        Name="test",
+        Description="A test parameter",
+        Value="value",
+        Type="String",
+        DataType="text",
```

Comment:
If DataType defaulfs to text, I don't think we need to change this invocation.
The `test_put_parameter` is modified so that we test that the parameter is supported, so all other tests do no need to pass the parameter - just the assertion that the default value is returned.

Reply:
Good call! Fixed!